### PR TITLE
Sample Data - Banner and Newsflash

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #3 .

### Summary of Changes
The purpose of these changes in sample data is giving testers and developers some data for testing chenges on cassiopeia.Different layouts, especially the module classes can be prsented

The first step in sample data were some addional menuItems to present a  drop down  menu ( metismenu)
This second steps adds makes more changes.

- Add some text and a button in the banner overlay the banner area.
- Add some intro images to articles
- Make the MenuItem "Blog" a Link to a Blog in mansonry layout
- Some modules   which are not < nav > are tagged as < aside >  (a11y) see  chart in comment https://github.com/joomla/cassiopeia/pull/55#issuecomment-685997907
- A **newflash module** has been added. There are three horizontal boxes in position top-a.
- These articles have an intro-image, at the moment it is the banner image.
- Blog featurd (Template) is in a two-columns layout with one leading article, using modifier classes for blog-layout..
- Newslash and banner are only on the homepage.

cassiopeia2

### Testing Instructions
Install blog sample data on a fresh installation. 
Make sure that you have the newest version and made npm ci for the horizontal layout of the newsflash boxes.

### Note
Sample blog data will be changed permanently to show important and attractive features, so will be updated during the whole development of cassiopeia.

### CODE REVIEW

**Comment by brianteeman** [comment]
> The section element is not a generic container element. When an element is needed only for styling purposes or as a convenience for scripting, authors are encouraged to use the div element instead. A general rule is that the section element is appropriate only if the element's contents would be listed explicitly in the document's outline.

Source: https://html.spec.whatwg.org/multipage/sections.html#the-section-element

Also see a longer explanation https://keithjgrant.com/posts/2018/03/html5-sectioning-and-landmark-elements/

http://html5doctor.com/avoiding-common-html5-mistakes/

https://www.w3.org/WAI/GL/wiki/Using_HTML5_section_element#The_.3Csection.3E_Element

Finally although the automated tests wont pick it up a section must have an accessible name otherwise it is meaningless to the document structure. A module without a title would be an obvious case for this to fail.

h5d-sectioning-flowchart

**Comment by chmst** [comment]
Thank you for this chart @brianteeman, very helpful.

File: administrator/language/en-GB/plg_sampledata_blog.ini [hunk 1]
```diff
@@ -72,7 +72,7 @@ PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_5_TITLE="Older Posts"
 PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_6_TITLE="Bottom Menu"
 PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_7_TITLE="Search"
 PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_8_TITLE="Image"
-PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_8_CONTENT="With Joomla you have chosen one of the most powerful CMS!"
+PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_8_CONTENT="<h1 class=\"display-3 text-thin\">Welcome to Joomla on Cassiopeia!</h1><p class=\"lead\">You have chosen one of the most powerful CMS Systems in the world.></p><p class=\"lead\">Cassiopeia is the  multi-purpose frontend template for Joomla 4.></p><p><a class=\"btn btn-warning btn-lg\" href=\"index.php\">Learn more »</a></p>"
```
Comment (brianteeman) [inline administrator/language/en-GB/plg_sampledata_blog.ini]:
```suggestion
PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_8_CONTENT="<h1 class=\"display-3 text-thin\">Welcome to Joomla on Cassiopeia!</h1><p class=\"lead\">You have chosen one of the most powerful CMS Systems in the world.></p><p class=\"lead\">Cassiopeia is the multi-purpose frontend template for Joomla 4.></p><p><a class=\"btn btn-warning btn-lg\" href=\"index.php\">Learn more »</a></p>"
```

remove extra space

**Comment by brianteeman** [comment]
Glad to have helped. This is why you can not rely on automated testing alone for accessibility.

Could you please update the original description

File: administrator/language/en-GB/plg_sampledata_blog.ini (hunk 1 above)
Comment (chmst) [reply administrator/language/en-GB/plg_sampledata_blog.ini]: There were also two > too much.

**Comment by chmst** [comment]
This step is ready for testing.

File: plugins/sampledata/blog/blog.php [hunk 2]
```diff
@@ -342,10 +372,14 @@ public function onAjaxSampledataApplyStep1()
 			$article['language']        = $language;
 			$article['associations']    = array();
 			$article['state']           = 1;
-			$article['images']          = '';
 			$article['metakey']         = '';
 			$article['metadesc']        = '';
 
+			if (!isset($article['images']))
+			{
+				$article['images']          = '';
```
Comment (richard67) [inline plugins/sampledata/blog/blog.php]:
```suggestion
				$article['images'] = '';
```

File: plugins/sampledata/blog/blog.php [hunk 3]
```diff
@@ -1059,7 +1121,7 @@ public function onAjaxSampledataApplyStep3()
 			array(
 				'title'     => Text::_('PLG_SAMPLEDATA_BLOG_SAMPLEDATA_MODULES_MODULE_11_TITLE'),
 				'ordering'  => 4,
-				'position'  => 'cpanel',
+				'position'  => 'cPanel',
```
Comment (richard67) [inline plugins/sampledata/blog/blog.php]: Is this change right? I'm not sure.

Comment (brianteeman) [reply plugins/sampledata/blog/blog.php]:
```suggestion
				'position'  => 'cpanel',
```

No change required
https://github.com/joomla/joomla-cms/blob/eeefb15c78e27a465362abccef64be09b9a449c2/administrator/templates/atum/templateDetails.xml#L43

**Comment by chmst** [comment]
Thanks for reviewing. I have changed the id 101 (Homepage) from hardcoded to getMenu()->getDefault().

File: plugins/sampledata/blog/blog.php [hunk 4]
```diff
@@ -881,6 +917,8 @@ public function onAjaxSampledataApplyStep3()
 					'format'       => 'rss',
 					'layout'       => '_:default',
 					'cache'        => 0,
+
+					'module_tag'   => 'section',
```
Comment (richard67) [inline plugins/sampledata/blog/blog.php]:
```suggestion
					'module_tag'   => 'section',
```

**Comment by chmst** [comment]
Unfortunately $app->getMenu()->getDefault() doesn't work in this context. I am on it for a solution

**Comment by richard67** [comment]
`$app->getMenu('site')->getDefault()` works, but not for multilanguage. When installing Blog sample data several times, one time for each language, all is fine except of the module's assignment to the home page only, whis is always the home for all languages in the Main Menu for all languages and not the language-specific home like it should be. Will see if I can help.

File: administrator/language/en-GB/plg_sampledata_blog.ini [hunk 5]
```diff
@@ -19,20 +19,20 @@ PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_ARTICLE_3_FULLTEXT="<p>On the full page y
...
-PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_ARTICLE_8_TITLE="Lorem ipsum3"
+PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_ARTICLE_6_FULLTEXT=" This will be a nice documentation on multi column Layouts in J4. <br><br>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est "
+PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_ARTICLE_6_INTROTEXT="<p>Millins of Websites are built on Joomla!</p>"
```
Comment (richard67) [inline administrator/language/en-GB/plg_sampledata_blog.ini]:
```suggestion
PLG_SAMPLEDATA_BLOG_SAMPLEDATA_CONTENT_ARTICLE_6_INTROTEXT="<p>Millions of Websites are built on Joomla!</p>"
```

**Review by richard67 — APPROVED** [review]

**Pull request merged**